src/main/kotlin/no/nav/familie/ef/sak/vilkår/VurderingService.kt

```diff
@@ -166,6 +170,102 @@ class VurderingService(
         tilbakestillEndretTidForKopierteVurderinger(kopiAvVurderinger, tidligereVurderinger)
     }
 
+    @Transactional
+    fun gjenbrukInngangsvilkårVurderinger(
+        nåværendeBehandlingId: UUID,
+        tidligereBehandlingId: UUID,
+    ) {
+        validerBehandlingForGjenbruk(
+            nåværendeBehandlingId,
+            tidligereBehandlingId
+        )
+        val forrigeVurderingIdTilNåværendeBarnMap =
+            finnBarnPåBeggeBehandlinger(tidligereBehandlingId, nåværendeBehandlingId)
+        val sivilstandErLik =
+            hentGrunnlagsdataOgValiderSivilstand(tidligereBehandlingId, nåværendeBehandlingId)
+        val (tidligereVurderinger, nåværendeVurderinger) = finnVurderingerSomSkalGjenbrukes(
+            tidligereBehandlingId,
+            sivilstandErLik,
+            forrigeVurderingIdTilNåværendeBarnMap,
+            nåværendeBehandlingId
+        )
+
+        val vurderingerSomSkalLagres = tidligereVurderinger.mapNotNull { tidligereVurdering ->
+            if (tidligereVurdering.type == VilkårType.ALENEOMSORG) {
+                val barnForVurdering = forrigeVurderingIdTilNåværendeBarnMap[tidligereVurdering.barnId]
+                val nåværendeVurdering =
+                    nåværendeVurderinger.firstOrNull { nåværendeVurdering -> nåværendeVurdering.type == tidligereVurdering.type  && nåværendeVurdering.barnId == barnForVurdering?.id }
+                if (nåværendeVurdering != null) {
+                    tidligereVurdering.copy(
+                        id = nåværendeVurdering.id,
+                        behandlingId = nåværendeBehandlingId,
+                        sporbar = nåværendeVurdering.sporbar,
+                        barnId = nåværendeVurdering.barnId,
+                    )
+                } else {
+                    null
+                }
+            } else {
+                val nåværendeVurdering = nåværendeVurderinger.first { b -> tidligereVurdering.type == b.type }
+                tidligereVurdering.copy(
+                    id = nåværendeVurdering.id,
+                    behandlingId = nåværendeBehandlingId,
+                    sporbar = nåværendeVurdering.sporbar,
+                )
+            }
+        }
```

Comment:
Mulig jeg tar feil her, men tror det skal bli riktig
På grunn av at `nåværendeVurdering.type == tidligereVurdering.type  && nåværendeVurdering.barnId == barnForVurdering?.id ` gir true når:
* Typen er lik, og barnId er lik
* For vilkår som ikke er Aleneomsorg så er barnId er null, så er de lik

```suggestion
        val vurderingerSomSkalLagres = tidligereVurderinger.mapNotNull { tidligereVurdering ->
            val barnForVurdering = forrigeVurderingIdTilNåværendeBarnMap[tidligereVurdering.barnId]
            nåværendeVurderinger.firstOrNull { it.type == tidligereVurdering.type && it.barnId == barnForVurdering?.id }
                ?.let {
                    tidligereVurdering.copy(
                        id = it.id,
                        behandlingId = behandlingId,
                        sporbar = it.sporbar,
                        barnId = it.barnId,
                    )
                }
        }
```

Reply:
Alternativt: Hvis det er viktig å dele opp de, så kanskje man burde endre uthenting av barnForVurdering, til 
```kotlin
val barnForVurdering = forrigeVurderingIdTilNåværendeBarnMap[tidligereVurdering.barnId] ?: return@mapNotNull null
```

og kan då fjerne `?` fra  `barnForVurdering?.id`, sånn at man alltid er sikker på at det er match på barnId for disse (då barnId ikke er null på barnForVurdering)
(men ikke begge disse 2 forslagen sammen, det blir feil)

Reply:
Her tenker jeg tester for både vilkår med og uten barn er fint å ha 🧪

Reply:
forenklet kode i [c593a9d](https://github.com/navikt/familie-ef-sak/pull/1740/commits/c593a9dea7933755dfa27eccbbbd10521a4ac189) 👍